v2/spiffetls/tlsconfig/config_test.go

```diff
@@ -263,10 +360,20 @@ func TestGetCertificate(t *testing.T) {
 		},
 	}
 
+	localTrace := tlsconfig.Trace{
+		GetCertificate: func() interface{} {
+			fmt.Printf("got start of GetTLSCertificate\n")
+			return nil
+		},
+		GotCertificate: func(interface{}, tlsconfig.GotCertificateInfo) {
+			fmt.Printf("got end of GetTLSCertificate\n")
+		},
+	}
+
 	for _, testCase := range testCases {
 		testCase := testCase
 		t.Run(testCase.name, func(t *testing.T) {
-			getCertificate := tlsconfig.GetCertificate(testCase.source)
+			getCertificate := tlsconfig.GetCertificate(testCase.source, tlsconfig.WithTrace(localTrace))
```

Comment:
It would be good to assert that `GetCertificate` and `GotCertificate` executed as expected. We may check what was written to stdout.

Reply:
an option is to create channels and push when we get/got certificates, so we can have more control there and assert expected resutls